firefox-ios/Client/Frontend/Browser/QRCodeViewController.swift

```diff
@@ -384,7 +384,7 @@ extension QRCodeViewController: AVCaptureMetadataOutputObjectsDelegate {
                let text = metaData.stringValue {
                 captureSession.stopRunning()
                 // Open QR codes only when they are recognized as webpages, otherwise open as text
-                if let url = URIFixup.getURL(text), url.isWebPage() {
+                if let url = URIFixup.getURL(text), url.isWebPage(), url.baseDomain != nil {
```

Comment:
This seems fine, can't really think of a time when we won't have any base domain